### PR TITLE
Send dialog: sizing and layout rules to limit dialog height jumping between states

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
#13377 
### What does the PR do

Applies the new sizing rules for SendModal as per design https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=21233-57676&mode=design&t=2O68lxNGG9g1b1tx-4

Please note even though the code looks like its been touched a lot, its only been rearranged or the container has been changed

### Affected areas

SendModal

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/a180ee29-5cfe-4c5d-859f-a3d8e9572a51

### CODE REVIEW

**Comment by dlipicar** [comment]
Is it just me or the background shade on the per-chain values looks weird?
image

I cannot find that in Figma actually

**Comment by Khushboo-dev-cpp** [comment]
> Is it just me or the background shade on the per-chain values looks weird? It seems not to be positioned properly. image
> 
> I cannot find that in Figma actually

Seems like thats a scrollbar on top of it which is making it look weird, its for scrolling

**Comment by Khushboo-dev-cpp** [comment]
> > Is it just me or the background shade on the per-chain values looks weird? It seems not to be positioned properly. image
> > I cannot find that in Figma actually
> 
> Seems like thats a scrollbar on top of it which is making it look weird, its for scrolling

However this is an issue that existed before my changes, we probably need to ask Ben and get designs on how to handle this small scrolling area here.

File: ui/imports/shared/popups/send/views/TokenListView.qml [hunk 1]
```diff
@@ -80,85 +80,69 @@ Item {
...
+            StatusListView {
+                id: tokenList
 
-                Component {
-                    id: communityInfoPopupCmp
-                    CommunityAssetsInfoPopup {}
+                Layout.fillWidth: true
+                Layout.fillHeight: true
+
+                header: root.browsingHoldingType === Constants.TokenType.ERC20 ? tokenHeader : collectibleHeader
```
Comment (dlipicar) [inline ui/imports/shared/popups/send/views/TokenListView.qml:136]: nit: same check done several times, could be replaced by some `isBrowsingTypeERC20` property defined once

**Review by dlipicar — APPROVED** [review]
Looks good! In case you haven't please check these changes don't break the relevant Storybook pages

**Comment by Khushboo-dev-cpp** [comment]
> Looks good! In case you haven't please check these changes don't break the relevant Storybook pages

thanks for the hint, TokenListViewPage was indeed broken. Ive fixed it now :)

File: ui/imports/shared/popups/send/views/SendModalFooter.qml [hunk 2]
```diff
@@ -30,6 +30,15 @@ Rectangle {
...
         samples: 17
         color: Theme.palette.dropShadow
+    }      
+
+    // This to have the square edges on top
+    Rectangle {
+      anchors.top: footer.top
+      anchors.left: footer.left
+      anchors.right: footer.right
+      color: Theme.palette.baseColor3
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/SendModalFooter.qml:40]: Just curious, how does this look in dark mode? ;)

File: ui/imports/shared/popups/send/SendModal.qml [hunk 3]
```diff
@@ -214,296 +215,268 @@ StatusDialog {
...
+                        formatCurrencyAmount: d.currencyStore.formatCurrencyAmount
                         onReCalculateSuggestedRoute: popup.recalculateRoutesAndFees()
-                        visible: recipientLoader.ready && !!d.selectedHolding && amountToSendInput.inputNumberValid
-                        errorType: d.errorType
+                        input.input.tabNavItem: recipientLoader.item
+                        Keys.onTabPressed: event.accepted = true
+                    }
+
+                    // Horizontal spacer
+                    RowLayout {}
```
Comment (caybro) [inline ui/imports/shared/popups/send/SendModal.qml:339]: That's a weird way to have a spacer

File: ui/imports/shared/popups/send/views/TabAddressSelectorView.qml [hunk 4]
```diff
@@ -46,44 +44,38 @@ Item {
...
+            anchors.fill: parent
 
             StatusListView {
                 id: savedAddresses
-                anchors.horizontalCenter: parent.horizontalCenter
-                anchors.top: parent.top
-                implicitWidth: parent.width
-                height: savedAddresses.contentHeight
+                Layout.fillWidth: true
+                Layout.fillHeight: true
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/TabAddressSelectorView.qml:78]: You don't have to, that's a default inside a `StackLayout`

File: ui/imports/shared/popups/send/views/TabAddressSelectorView.qml [hunk 5]
```diff
@@ -115,21 +107,13 @@ Item {
...
             StatusListView {
                 id: myAccounts
                 objectName: "myAccountsList"
-                anchors.horizontalCenter: parent.horizontalCenter
-                anchors.top: parent.top
-                width: parent.width
-                height: myAccounts.contentHeight
+
+                Layout.fillWidth: true
+                Layout.fillHeight: true
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/TabAddressSelectorView.qml:116]: Same here, not needed

File: ui/imports/shared/popups/send/views/TabAddressSelectorView.qml [hunk 6]
```diff
@@ -147,43 +131,12 @@ Item {
...
 
             StatusListView {
                 id: recents
-                anchors.horizontalCenter: parent.horizontalCenter
-                anchors.top: parent.top
-                width: parent.width
-                height: recents.contentHeight
+
+                Layout.fillWidth: true
+                Layout.fillHeight: true
```
Comment (caybro) [inline ui/imports/shared/popups/send/views/TabAddressSelectorView.qml:139]: Ditto

**Review by caybro — APPROVED** [review]
Looks good, just some minor stuff

**Pull request merged**